noobaa/noobaa-core
# Prometheus Initial Integration

### Explain the changes
#### 1. New Features / Capabilities (Sync with Liran):
- Prometheus native integration - Stage I 
    - Added prom-client dependency
    - Added Prometheus Reporter singleton
    - Added web server routes for scraping
    - Current metrics  supported
        - Object sizes in the system
        - Cloud resource usage bandwidth and ops

#### 3. Other Changes:
- .nvmrc updated node to 10.15.0

### Testing Instructions:
- Check the routes /metrics and /metrics/counter
- Install prometheus (container avaialble) and update its config to scrape the server, verify the noobaa values are visible through it

### Reminders:
- Create a new test - the first is the hardest
- User Experience is top priority
- Design for failure
- Keep it simple
- `npm test`
- Imagine a support call - Add diagnostics info, phonehome info, activity log events, external syslog
- Upgrade - DB schema, server platform, agents install, npm packages
